lib/widgets/mixins/get_preview_url_mixin.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/domain/app_state/preview_url/get_preview_url_success.dart';
+import 'package:fluffychat/domain/usecase/preview_url/get_preview_url_interactor.dart';
+import 'package:flutter/material.dart';
+import 'package:matrix/matrix.dart';
+
+mixin GetPreviewUrlMixin {
+  static const int _defaultPreferredPointInTime = 2000;
+
+  final GetPreviewURLInteractor _getPreviewURLInteractor =
+      getIt.get<GetPreviewURLInteractor>();
+
+  final getPreviewUrlStateNotifier =
+      ValueNotifier<Either<Failure, Success>>(Right(GetPreviewUrlInitial()));
+
+  abstract String debugLabel;
+
+  void getPreviewUrl({
+    required Uri uri,
+    int? preferredPreviewTime,
+  }) {
+    _getPreviewURLInteractor
+        .execute(
+          uri: uri,
+          preferredPreviewTime:
+              preferredPreviewTime ?? _defaultPreferredPointInTime,
+        )
+        .listen(
+          _handleGetPreviewUrlOnData,
+          onError: _handleGetPreviewUrlOnError,
+          onDone: _handleGetPreviewUrlOnDone,
+        );
+  }
+
+  void _handleGetPreviewUrlOnData(Either<Failure, Success> event) {
+    Logs().d('$debugLabel::_handleGetPreviewUrlOnData()');
```

Comment:
i understand debuglabel, should change the name for easier understand

Reply:
this is the new way for debug in mixin
The class should override `debugLabel` so that the mixin can distinguish which class is using that mixin, in case 2 classes are logging the same mixin.